tests/test_pipelines_common.py

```diff
@@ -77,12 +77,17 @@ def get_tiny_config_from_class(configuration_class):
     model_tester = model_tester_class(parent=None)
 
     if hasattr(model_tester, "get_pipeline_config"):
-        return model_tester.get_pipeline_config()
+        config = model_tester.get_pipeline_config()
     elif hasattr(model_tester, "get_config"):
-        return model_tester.get_config()
+        config = model_tester.get_config()
     else:
+        config = None
         logger.warning(f"Model tester {model_tester_class.__name__} has no `get_config()`.")
 
+    if hasattr(model_tester, "update_config_with_model_class"):
+        config = model_tester.update_config_with_model_class(config, model_architecture)
```

Comment:
Would be nice to add a comment here

Reply:
Would this one do ?

```python
   # Perceiver is a bit specific since `d_model` needs to be defined
        # ahead of time in the config, but the actual desired `d_model` might
        # depend on the model class we want to use so this is an escape hatch
        # to enable adjusting `d_model` after we know which head we intend to
        # use
``` (This is the one that is in the method itself).